vertx-web-graphql/src/main/java/io/vertx/ext/web/handler/graphql/impl/ApolloWSConnectionHandler.java

```diff
@@ -94,22 +101,62 @@ private void handleMessage(JsonObject jsonObject) {
       return;
     }
 
-    ApolloWSMessage message = new ApolloWSMessageImpl(serverWebSocket, type, jsonObject);
+    ApolloWSMessageImpl message = new ApolloWSMessageImpl(serverWebSocket, type, jsonObject);
 
     Handler<ApolloWSMessage> mh = apolloWSHandler.getMessageHandler();
     if (mh != null) {
       mh.handle(message);
     }
 
+    Handler<ApolloWSConnectionInitEvent> connectionInitHandler = apolloWSHandler.getConnectionInitHandler();
+
     switch (type) {
       case CONNECTION_INIT:
-        connect();
+        if (!connectionInitialized.compareAndSet(false, true)) {
+          sendMessage(opId, ERROR, "CONNECTION_INIT can only be sent once");
```

Comment:
In this case, shouldn't we terminate the connection?

Reply:
Right, probably it would be better to close the connection if the client misbehaves, implemented it